rust/nasl-builtin-regex/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+use nasl_builtin_utils::{error::FunctionErrorKind, NaslFunction};
+use nasl_builtin_utils::{Context, Register};
+use nasl_function_proc_macro::nasl_function;
+use nasl_syntax::NaslValue;
+use regex::RegexBuilder;
+
+/// Matches a string against a regular expression.
+/// - string  String to search the pattern in
+/// - pattern the pattern that should be matched
+/// - icase   case insensitive flag
+/// - rnul    replace the null char in the string. Default TRUE.
+/// - multiline Is FALSE by default (string is truncated at the first
+///                “end of line”), and can be set to TRUE for multiline search.
+/// Return true if matches, false otherwise
+#[nasl_function(named(string, pattern, icase, rnul, multiline))]
+fn ereg(
+    string: NaslValue,
+    pattern: NaslValue,
+    icase: Option<bool>,
+    rnul: Option<bool>,
+    multiline: Option<bool>,
+) -> Result<NaslValue, FunctionErrorKind> {
```

Comment:
You can change the return type to `Result<bool, FunctionErrorKind>` and simply return a bool.